### PR TITLE
cohere: Fix cohere rerank

### PR DESCRIPTION
**Pull request by billytrend-cohere**
Fix cohere rerank inspired by https://github.com/langchain-ai/langchain/pull/19486

### CODE REVIEW

File: libs/partners/cohere/langchain_cohere/rerank.py
Comment (baskaryan) [inline libs/partners/cohere/langchain_cohere/rerank.py:1]: do we want to make this backwards compatible with v4?

Comment (baskaryan) [reply libs/partners/cohere/langchain_cohere/rerank.py:1]: ah this is to langchain-cohere, ignore me

**Review by baskaryan — APPROVED** [review]

File: libs/partners/cohere/tests/integration_tests/test_rerank.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+def test_langchain_cohere_rerank_documents() -> None:
+    """Test cohere rerank."""
+    rerank = CohereRerank()
+    test_documents = [
+        Document(page_content="This is a test document."),
+        Document(page_content="Another test document."),
+    ]
+    test_query = "Test query"
+    results = rerank.rerank(test_documents, test_query)
+    assert len(results) == 2
```
Comment (mspronesti) [inline libs/partners/cohere/tests/integration_tests/test_rerank.py:16]: How about creating a list of 4 (or more) docs and asserting the length of the results to be equal to 3 (default `n`)?

Comment (billytrend-cohere) [reply libs/partners/cohere/tests/integration_tests/test_rerank.py:16]: will follow up @mspronesti thanks!

Comment (mspronesti) [reply libs/partners/cohere/tests/integration_tests/test_rerank.py:16]: You can click on "commit suggestion" if that's faster :)

```suggestion
    test_documents = [
        Document(page_content="This is a test document."),
        Document(page_content="Another test document."),
        Document(page_content="Yet Another test document."),
        Document(page_content="Fourth test document."),
    ]
    test_query = "Test query"
    results = rerank.rerank(test_documents, test_query)
    assert len(results) == 3
```

**Pull request merged**